### DISCUSSION TITLE
Possibility of producing a periodic mesh with hex elements

### DISCUSSION THREAD

**Discussion by abarun22** [neperfepx/neper#729] [Q&A]
Hi all,
I am looking at creating a 'tet' mesh that exhibits periodicity by setting the periodicity option (to all,1) in the tessellation module. Not sure if this is a reasonable expectation, as you may never be able to produce collinear nodes in the periodic boundaries with 'tet' elements. What type of element is advised, if one wants to run an FE problem of my microstructure with periodicity conditions enforced?
Kind regards,
Arun

**Comment by rquey** [comment]
Only tet elements can be used to produce a periodic mesh.  Quick example:
```
neper -T -n 500 -per 1
neper -M n500-id1.tess -pl 100 
neper -V n500-id1.tess,n500-id1.msh -showelt1d all -dataelt1drad 0.004 -cameracoo x:-7:z -print d
```

![image](https://github.com/neperfepx/neper/assets/2659376/9d21c4f6-dc08-4b90-9761-ad196c77ea7a)
